planner/core/point_get_plan.go

```diff
@@ -600,13 +618,38 @@ func newBatchPointGetPlan(
 			}
 			handles[i] = kv.IntHandle(intDatum.GetInt64())
 			handleParams[i] = con
+			pairs := []nameValuePair{{colName: handleCol.Name.L, colFieldType: item.GetType(), value: *intDatum, con: con}}
+			if tbl.GetPartitionInfo() != nil {
+				tmpPartitionDefinition, _, isTableDual := getPartitionInfo(ctx, tbl, pairs)
+				if isTableDual {
+					return nil
+				}
+				if tmpPartitionDefinition != nil {
+					partitionInfos = append(partitionInfos, tmpPartitionDefinition)
+				}
+			}
 		}
+
+		sort.SliceStable(partitionInfos, func(i, j int) bool {
+			return partitionInfos[i].ID < partitionInfos[j].ID
```

Comment:
So this would sort by partition-id (or "global table id"), which would not be as expected, it would most likely be in the order of creation, so if one have done some partition management, like `EXCHANGE PARTITION` the order could be non-expected. Like:
```
create table tt (a int);
create table tp (a int) partition by range (a) (partition p1 values less than (1000000), partition p2 values less than (2000000));
-- Now it would show the partitions in order p1,p2
alter table tp exchange partition p2 with table tt;
-- Now it would show the partitions in order p2,p1
```
I would expect the order to be the same as when listing the partitions shown in `SHOW CREATE TABLE`.